File: JavaScript/Bit Manipulation/338. Counting Bits.js
/* 338. Counting Bits
Difficulty: Medium
Submiission Link: https://leetcode.com/problems/counting-bits/submissions/1385760283/ */

const countBits = (n) => {
    let result = Array(n + 1).fill(0);
    let offset = 1;
    for (let i = 1; i < n + 1; i++) {
      if (offset * 2 === i) {
        offset = i;
      }
      result[i] = 1 + result[i - offset];
    }
  
    return result;
  };

File: C#/Bit Manipulation/338. Counting Bits.cs
/* 338. Counting Bits
Difficulty: Medium
Submiission Link: https://leetcode.com/problems/counting-bits/submissions/1037892468/ */

int* countBits(int n, int* returnSize){
    *returnSize = n + 1;
    int* ret = (int*) malloc(*returnSize * sizeof(int));
    ret[0] = 0;
    for (int i = 1; i < *returnSize; i++) {
        ret[i] = (i & 1) + ret[i >> 1];
    }
    return ret;
}